File: modules/agent/agents/junior.go
package agents

import (
	"github.com/woyin/OrangeCoding/modules/ai"
	"github.com/woyin/OrangeCoding/modules/core"
	"github.com/woyin/OrangeCoding/modules/tools"
)

// NewJunior creates the simple task handling agent.
// RoleCoder, tools: [bash, read_file, write_file].
func NewJunior(provider ai.AiProvider, registry *tools.ToolRegistry, workDir string) *BaseAgent {
	allowedTools := []string{"bash", "read_file", "write_file"}
	return newFilteredAgent(provider, registry, workDir, core.RoleCoder, allowedTools,
		"You are Junior, the simple task handler. You handle straightforward tasks quickly and efficiently without overthinking.")
}

File: packages/agent/src/agents/junior.ts
/**
 * Junior - simple task handling agent.
 * Ported from modules/agent/agents/junior.go.
 */

import type { AiProvider } from "@orangecoding/ai";
import { AgentRole } from "@orangecoding/core";
import { ToolRegistry } from "@orangecoding/tools";
import { BaseAgent } from "./base.js";
import { newFilteredAgent } from "./helpers.js";

export function newJunior(provider: AiProvider, registry: ToolRegistry, workDir: string): BaseAgent {
  return newFilteredAgent(provider, registry, workDir, AgentRole.Coder,
    ["bash", "read_file", "write_file"],
    "You are Junior, the simple task handler. You handle straightforward tasks quickly and efficiently without overthinking.");
}
